src/CommunityPatch/Patches/EarlyStoryVisibleTimeoutPatch.cs

```diff
@@ -2,112 +2,123 @@
 
-    private void SetStoryVisibleTimeoutIfNeeded(QuestBase quest) {
+    private static void SetStoryVisibleTimeoutIfNeeded(QuestBase quest) {
       if (!IsFirstStoryPhase() || !quest.IsSpecialQuest || !quest.IsOngoing)
         return;
 
       // set visible timeout to be when vanilla would have (silently) timed 
       // out the quest,  minus a day to make very sure the quest doesn't 
       // somehow trigger vanilla's silent timeout too early.
-      CampaignTime newDueTime = FirstPhase.Instance.FirstPhaseStartTime
-                                  + CampaignTime.Years(FirstPhaseTimeLimitInYears)
-                                  - CampaignTime.Days(1);
-      if (quest.QuestDueTime != newDueTime) {
-        quest.ChangeQuestDueTime(newDueTime);
-        ShowNotification(new TextObject("{=QuestTimeRemainingUpdated}Quest time remaining was updated."), "event:/ui/notification/quest_update");
-      }
-    }
+      var newDueTime = GetFirstPhaseFirstPhaseStartTime(FirstPhaseInstance)
+        + CampaignTime.Years(FirstPhaseTimeLimitInYears)
+        - CampaignTime.Days(1);
+      if (quest.QuestDueTime == newDueTime)
+        return;
 
-    private static void ShowNotification(TextObject message, string soundEventPath = "") {
-      InformationManager.AddQuickInformation(message, 0, null, soundEventPath);
+      quest.ChangeQuestDueTime(newDueTime);
+      ShowNotification(
+        new TextObject("{=QuestTimeRemainingUpdated}Quest time remaining was updated."),
+        "event:/ui/notification/quest_update"
+      );
     }
 
-    class NextHourlyTickListener {
-      private Action<QuestBase> _func;
-      private QuestBase _quest;
-
-      public NextHourlyTickListener(Action<QuestBase> func, QuestBase quest) {
-        _func = func;
-        _quest = quest;
-        CampaignEvents.HourlyTickEvent.AddNonSerializedListener(this, OnHourlyTick);
-      }
-
-      private void OnHourlyTick() {
-        _func.Invoke(_quest);
-        CampaignEvents.HourlyTickEvent.ClearListeners(this);
-      }
-    };
+    private static void ShowNotification(TextObject message, string soundEventPath = "")
+      => InformationManager.AddQuickInformation(message, 0, null, soundEventPath);
 
     private void OnQuestStarted(QuestBase quest) {
+      // give OnHourlyTick's closure a late bound self-ref
+      var onHourlyTickBoxed = new StrongBox<Action>();
+
       // defer our updates until some time has passed, because they depend on whether 
       // FirstPhase or SecondPhase is active, and story Phase information is only 
       // updated after all OnQuestStarted handlers have fired.
-      new NextHourlyTickListener(new Action<QuestBase>(SetStoryVisibleTimeoutIfNeeded), quest);
+      onHourlyTickBoxed.Value = () => {
+        SetStoryVisibleTimeoutIfNeeded(quest);
+        CampaignEvents.HourlyTickEvent.ClearListeners(onHourlyTickBoxed.Value);
+      };
+
+      CampaignEvents.HourlyTickEvent.AddNonSerializedListener(this, onHourlyTickBoxed.Value);
```

Comment:
IIRC, the parameter to `ClearListeners` needs to be the same as the first param to the corresponding `AddNonSerializedListener` call

Reply:
See below.